src/Webcam.cpp

```diff
@@ -4,33 +4,48 @@
 
 #include "Webcam.hpp"
 
-#include <QFile>
+#include <memory>
 #include <iostream>
+
+#include <QFile>
 #include <QCameraInfo>
 #include <QCameraImageCapture>
 
+
 Webcam::Webcam() {
+    cams = QCameraInfo::availableCameras();
 
-    if(QCameraInfo::availableCameras().count() > 0) {
-        cam = new QCamera(QCameraInfo::availableCameras()[0]);
-        cam->setCaptureMode(QCamera::CaptureStillImage);
-        cam->start();
+    if (!cams.count())
+        return;
 
-        imgCapture = new QCameraImageCapture(cam);
+    cam = std::make_shared<QCamera>(cams[0]);
 
-        QObject::connect(imgCapture, &QCameraImageCapture::imageSaved, this, &Webcam::emitImage);
-    }
-    else {
-        cam = nullptr;
+    if (cam->isCaptureModeSupported(QCamera::CaptureStillImage)) {
+        cam->setCaptureMode(QCamera::CaptureStillImage);
+    } else {
+        std::cerr << "Camera does not support still image capture" << std::endl;
+        cam.reset();
+        return;
     }
+
+    cam->start();
+
+    imgCapture = std::make_shared<QCameraImageCapture>(cam.get());
+    std::cerr << "Capture to buffer supported: "
```

Comment:
Are you going to do something with this? If not, remove.

Reply:
Not right now, we'll need that as soon as we support capture to buffer. Anyway, it would be nice to know which platforms support capturing to a buffer.

Reply:
the log is ok by me, but it shouldn't be logged to cerr